### PR TITLE
RMT: test cleanups

### PR DESCRIPTION
**Pull request by wisp3rwind**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [ ] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [ ] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [ ] I have added necessary changes to user code to the latest [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal).
- [x] My changes are in accordance to the [esp-rs developer guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/DEVELOPER-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

### Pull Request Details 📖
RMT HIL tests have grown quite a bit through my recent PRs, and there's quite a bit of boilerplate. This PR doesn't actually manage to reduce the line count of the test module, but it should make it more readable, remove a few redundant tests, and add a test for differently structured data (with end marker in the second pulse code field; just to ensure that there's no relevant difference in hardware behavior in that case).

There's one unrelated comment that fixes an oversight in updating the documentation from #4049.

Overall, this touches quite a few lines. However, each commit only performs a single change, so this should be fairly easy to review commit-by-commit.

Only refactors the tests, so no changelog required.

#### Testing
Describe how you tested your changes.

### CODE REVIEW

File: hil-test/src/bin/rmt.rs [hunk 1]
```diff
@@ -860,18 +957,19 @@ mod tests {
...
 
-        let (mut tx_channel, mut rx_channel) = ctx.setup_loopback(&tx_config, &rx_config);
+        let (mut tx_channel, mut rx_channel) = ctx.setup_loopback(&conf);
 
-        let tx_data: [_; TX_COUNT + 1] = generate_tx_data(false, true);
-        let mut rx_data: [PulseCode; MAX_RX_LEN] = [Default::default(); MAX_RX_LEN];
-        let mut expected_data: [PulseCode; MAX_RX_LEN] = [Default::default(); MAX_RX_LEN];
+        let tx_data = generate_tx_data(&conf);
+        let mut rx_data = vec![PulseCode::default(); MAX_RX_LEN];
+        let mut expected_data = vec![PulseCode::default(); MAX_RX_LEN];
```
Comment (bugadani) [inline hil-test/src/bin/rmt.rs:972]: These don't have to be heap-allocated, do they?

Comment (wisp3rwind) [reply hil-test/src/bin/rmt.rs:972]: No, I think I simply changed all data buffers to be on the heap in this PR. It doesn't really matter for this test, and I don't have a strong opinion about it. I can change this back to being on the stack if you'd prefer that.

Part of the reason for the change to `esp-alloc` (aside from the const generics being inconvenient) was to simplify adding tests with a wider range of data sizes in order to test ensure that there are no bugs for edge cases around buffer wrapping. However, I didn't actually implement that in this PR, yet, and it wouldn't be relevant for this specific test anyway.
